pkg/container/device.go

```diff
@@ -163,3 +165,32 @@ func (c *container) specAddContainerConfigDevices() error {
 	}
 	return nil
 }
+
+func disableDeviceOwnershipFromSecurityContext(annotations map[string]string) bool {
+	if v, ok := annotations[crioann.DisableDeviceOwnershipFromSecurityContextAnnotation]; ok {
+		if v == "true" {
+			return true
+		}
+	}
+	return false
+}
+
+func getDeviceUID(config *types.ContainerConfig, hostUID uint32) *uint32 {
+	if userval := config.Linux.SecurityContext.RunAsUser; userval != nil {
+		uid := uint32(userval.Value)
+		if uid > 0 && !disableDeviceOwnershipFromSecurityContext(config.Annotations) {
+			return &uid
+		}
+	}
+	return &hostUID
+}
+
+func getDeviceGID(config *types.ContainerConfig, hostGID uint32) *uint32 {
+	if groupval := config.Linux.SecurityContext.RunAsGroup; groupval != nil {
+		gid := uint32(groupval.Value)
+		if gid > 0 && !disableDeviceOwnershipFromSecurityContext(config.Annotations) {
```

Comment:
this value will never change, so we should call disableDeviceOwnershipFromSecurityContext once and pass that value into these functions

Reply:
done